src/util/numeric.h

```diff
@@ -416,4 +416,9 @@ inline u32 npot2(u32 orig) {
 	return orig + 1;
 }
 
+// http://stackoverflow.com/questions/2249110/how-do-i-make-a-portable-isnan-isinf-function
+// Let's just use our own macro like Lua does internally. At least it works
+// reliably and is portable...
+#define util_isnan(a) (!((a)==(a)))
```

Comment:
Operators need to have a space in between variables.  Macros should be CAPITAL_LETTERS_WITH_UNDERSCORES.
Shouldn't you make the prefix "MY" to fit in with the other macros redefined from the standard, i.e. MYISNAN()?

Reply:
You could use the optimize attribute or something similar (e.g. `#ifdef __FAST_MATH__`) to ensure that util_isnan works correctly.
If someone compiles minetest using gcc with --ffast-math (included in -Ofast), the nan test doesn't work otherwise.